debugging.md

```diff
@@ -0,0 +1,52 @@
+# Debugging Go based operator
+
+## Debugging operator in vscode
+
+### Prerequisite
+
+* install [current used go version](https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/vars/default.yaml)
+* install [current used operator-sdk version](https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/vars/default.yaml)
+* install vscode
+* make sure [go extension for vscode](https://marketplace.visualstudio.com/items?itemName=golang.Go)
+is installed
+* install [delve debugger for go](https://github.com/go-delve/delve)
+```bash
+go install github.com/go-delve/delve/cmd/dlv@latest
+```
+* Deploy the needed operator CRDs/manifests either manually or install and uninstall via olm
+```bash
+oc login -u kubeadmin -p '12345678' https://api.ostest.test.metalkube.org:6443
+make install
+```
+
+### Run debugger via vscode
+A `.vscode/launch.json` needs to be created which instructs vscode how to start the debugger. Create the `.vscode/launch.json` with `Run` -> `Add configurarion` -> `Go: Launch Package`
+```json
+{
+        "version": "0.2.0",
+        "configurations": [
+        
+                {
+                        "name": "Launch Package",
+                        "type": "go",
+                        "request": "launch",
+                        "mode": "auto",
+                        "program": "<path to operator base dir>",
```

Comment:
Maybe we could also add a task and add a line:
`            "preLaunchTask": "Run make",`
and the respective tasks.json. I use something like that:
`{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Run make",
            "type": "shell",
            "command": "make install manifests generate fmt vet",
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "presentation": {
                "reveal": "always",
                "panel": "new"
            },
            "options": {
                "cwd": "<path to the operator base dir>",
                "env": {
                    "OPERATOR_TEMPLATES": "./templates",
                    "PATH": "~/.crc/bin/oc:${env:PATH}"
                }
            },
            "problemMatcher": []
        }
    ]
  }
`

Reply:
thats a good idea, will add it

Reply:
updated . its now generic that you won't have to specify the paths